Makefile.inc

```
@@ -142,7 +142,19 @@ lint: gitlint golangci-lint markdownlint yamllint
 
 # [markdownlint] validates Markdown files in the project
 markdownlint:
-	markdownlint markdownlint -c .markdownlint.yml -i vendor .
+	md_ignored=""; \
+	if [ -r .mdignore ]; then \
+		while read -r ignore_line; do \
+			echo $(ignore_line) ; \
+			if [ -r $(ignore_line) ]; then \
+				md_ignored="$(md_ignored) --ignore $(line)"; \
+			fi \
+		done < .mdignore ; \
+		echo $(md_ignored) ; \
+		markdownlint -c .markdownlint.yml $(md_ignored) . ; \
+	else \
+		markdownlint -c .markdownlint.yml . ; \
+	fi
```

I haven’t run this through `shellcheck`, but I have checked it works:

```suggestion
	md_ignored=(vendor); \
	if [ -r .mdignore ]; then \
		md_ignored+=($$(< .mdignore)); \
	fi; \
	markdownlint -c .markdownlint.yml $${md_ignored[@]/#/-i }
```

---

> 			if [ -r $(ignore_line) ]; then \

The thought with this extra bit was to verify that each chunk we read from .mdignore and pass to markdownlint is a valid path to a file, to avoid opening up a direct path to the shell and mitigate potential attacks with crafted .mdignore lines. That may not be worth the effort/complexity though.